ledger-core/virtual/execute/execute.go

```diff
@@ -311,10 +311,6 @@ func (s *SMExecute) stepTakeLock(ctx smachine.ExecutionContext) smachine.StateUp
 	}
 
 	if ctx.Acquire(executionSemaphore).IsNotPassed() {
-		if s.isConstructor {
```

Comment:
why you've removed this? how could you not acquire semaphor on object contruction?

Reply:
Please re-read that part and apply it to deduplication+AwaitDelegate